doctr/utils/reconstitution.py

```diff
@@ -13,61 +14,156 @@
 __all__ = ["synthesize_page", "synthesize_kie_page"]
 
 
+def _warn_rotation(entry: Dict[str, Any], already_warned: bool) -> bool:  # pragma: no cover
+    if len(entry["geometry"]) == 4 and not already_warned:
+        logging.warning("Polygons with larger rotations will lead to inaccurate rendering")
+        return True
+    return already_warned
```

Comment:
`already_warned` should be a global variable, not a parameter. It's not the responsability of other callers to check if warning has been already raised

Something like this:
```suggestion
ROTATION_WARNING = False

def _warn_rotation(entry: Dict[str, Any]) -> None:
    if not ROTATION_WARNING and len(entry["geometry"]) == 4:
        logging.warning("Polygons with larger rotations will lead to inaccurate rendering")
        ROTATION_WARNING = True
```

Reply:
Yeah good catch will be applied tomorrow :)